test_pytest/test_cartesian.py

```diff
@@ -5,15 +5,17 @@
 import splinepy
 
 
-def test_cartesian_product():
+def test_cartesian_product(request):
     """Test cartesian product test"""
 
+    rng = request.getfixturevalue("np_rng")
```

Comment:
```suggestion
def test_cartesian_product(np_rng):
    """Test cartesian product test"""
```

why do you get it via request? You just need to do it if you have only access to the fixture via strings (I used this only in conjunction with `parameterize`)

You need to change it later down as well

Reply:
I'll change it, thanks.